contracts/VaultFactory.vy

```diff
@@ -116,8 +145,47 @@ def set_protocol_fee_recipient(new_protocol_fee_recipient: address):
     @param new_protocol_fee_recipient The new protocol fee recipient
     """
     assert msg.sender == self.governance, "not governance"
-    log UpdateProtocolFeeRecipient(self.protocol_fee_config.fee_recipient, new_protocol_fee_recipient)
-    self.protocol_fee_config.fee_recipient = new_protocol_fee_recipient
+    log UpdateProtocolFeeRecipient(self.default_protocol_fee_config.fee_recipient, new_protocol_fee_recipient)
+    self.default_protocol_fee_config.fee_recipient = new_protocol_fee_recipient
+
+@external
+def set_custom_protocol_fee_bps(vault: address, custom_protocol_fee: uint16):
+    """
+    @notice Allows Governance to set custom protocol fees
+    for a specific vault or strategy.
+    @param vault The address of the vault or strategy to customize.
+    @param custom_protocol_fee The custom protocol fee in BPS.
+    """
+    assert msg.sender == self.governance, "not governance"
+    assert custom_protocol_fee <= MAX_FEE_BPS, "fee too high"
+
+    self.custom_protocol_fee_config[vault] = PFConfig({
+        fee_bps: custom_protocol_fee,
+        fee_last_change: convert(block.timestamp, uint32),
+        fee_recipient: self.default_protocol_fee_config.fee_recipient
```

Comment:
i can't think of a scenario where needed really, but may as well allow a custom recipient?